src/main/kotlin/com/frcteam3636/frc2024/subsystems/arm/ArmIO.kt

```diff
@@ -17,188 +18,250 @@ import edu.wpi.first.units.Voltage
+}
 
 
-    class ArmIOReal: ArmIO{
+class ArmIOReal: ArmIO {
 
-        private val leftMotor = TalonFX(CTREDeviceId.LeftArmMotor)
+    private val leftMotor = TalonFX(CTREDeviceId.LeftArmMotor)
 
-        private val rightMotor = TalonFX(CTREDeviceId.RightArmMotor)
+    private val rightMotor = TalonFX(CTREDeviceId.RightArmMotor)
 
-        private val absoluteEncoder = DutyCycleEncoder(DigitalInput(8))
+    private val leftAbsoluteEncoder = DutyCycleEncoder(DigitalInput(0))
+    private val rightAbsoluteEncoder = DutyCycleEncoder(DigitalInput(1))
 
-        override fun updateInputs(inputs: ArmInputs) {
-            inputs.position = Rotations.of(absoluteEncoder.absolutePosition)
-            inputs.absoluteEncoderConnected = absoluteEncoder.isConnected
+    init {
+        Logger.recordOutput("/Arm/CoastMode", false)
+    }
 
-            inputs.absoluteEncoderPosition = Rotations.of(absoluteEncoder.absolutePosition)
+    override fun updateInputs(inputs: ArmInputs) {
+        val offsetlessLeftPosition = Rotations.of(-leftAbsoluteEncoder.get() * CHAIN_GEAR_RATIO)
+        inputs.leftPosition = offsetlessLeftPosition + LEFT_ZERO_OFFSET
+        Logger.recordOutput("/Arm/Required Left Offset", offsetlessLeftPosition.negate())
 
-            inputs.leftPosition = Rotations.of(leftMotor.position.value)
-            inputs.leftVelocity = RotationsPerSecond.of(leftMotor.velocity.value)
+        val offsetlessRightPosition = Rotations.of(rightAbsoluteEncoder.get() * CHAIN_GEAR_RATIO)
+        inputs.rightPosition = offsetlessRightPosition + RIGHT_ZERO_OFFSET
+        Logger.recordOutput("/Arm/Required Right Offset", offsetlessRightPosition.negate())
 
+        inputs.leftRelativePosition = Rotations.of(leftMotor.position.value)
+        inputs.rightRelativePosition = Rotations.of(rightMotor.position.value)
 
-            inputs.rightPosition = Rotations.of(rightMotor.position.value)
-            inputs.rightVelocity = RotationsPerSecond.of(rightMotor.position.value)
+        inputs.leftAbsoluteEncoderConnected = leftAbsoluteEncoder.isConnected
+        inputs.rightAbsoluteEncoderConnected = rightAbsoluteEncoder.isConnected
 
-            inputs.rightCurrent = Volts.of(leftMotor.motorVoltage.value)
+        inputs.leftVelocity = RotationsPerSecond.of(leftMotor.velocity.value)
+        inputs.rightVelocity = RotationsPerSecond.of(rightMotor.velocity.value)
 
-            leftMotor.setPosition(inputs.absoluteEncoderPosition.`in`(Rotations))
-            rightMotor.setPosition(inputs.absoluteEncoderPosition.`in`(Rotations))
-        }
+        inputs.leftCurrent = Volts.of(leftMotor.motorVoltage.value)
+        inputs.rightCurrent = Volts.of(rightMotor.motorVoltage.value)
 
-        override fun updatePosition(position: Measure<Angle>) {
-            leftMotor.setPosition(position.`in`(Rotations))
-            rightMotor.setPosition(position.`in`(Rotations))
-        }
+        inputs.reference = Rotations.of(leftMotor.closedLoopReference.value)
+    }
```

Comment:
Given our history with getting funky encoder values we should be doing sanity checking here. Not only the encoder position but also the relative position and voltage.